r2-core/src/main/java/com/linkedin/r2/transport/http/client/SmoothRateLimiter.java

```diff
@@ -107,7 +133,17 @@ public void submit(Callback<None> callback) throws RejectedExecutionException
 
     if (_pendingCount.get() >= _maxBuffered)
     {
-      throw new RejectedExecutionException("Cannot submit callback because the buffer is full at " + _maxBuffered);
+      if (_bufferOverflowMode == BufferOverflowMode.DROP)
+      {
+        throw new RejectedExecutionException(
+          String.format("PEGA_2000: Cannot submit callback because the buffer is full at %d tasks for ratelimiter: %s", _maxBuffered, _rateLimiterName));
+      }
+      else
+      {
+        _rateLimitedLoggerOverBuffer.error(String.format(
+          "PEGA_2001: the buffer is full at %d tasks for ratelimiter: %s. Executing a request immediately to avoid overflowing and dropping the task.", _maxBuffered,
```

Comment:
Should this be a warning instead? And "PEGA_xxx" is normally only returned with exception right? Without the exception I don't see why we want to give this code in a rate limited log

Reply:
In theory, this is a "Serious problem" that they should solve, so I made it an error. But I can make this a warning if you prefer